From: Al Viro <viro@zeniv.linux.org.uk>
Date: Thu, 3 Apr 2014 22:44:19 -0400
Subject: [PATCH] ceph_sync_{,direct_}write: fix an oops on
 ceph_osdc_new_request() failure
Git-commit: eab87235c0f5979503a547f836a93a3d327c4201
Patch-mainline: v3.15
References: Fate#318586


ceph_osdc_put_request(ERR_PTR(-error)) oopses.  What we want there
is break, not goto out.

Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/file.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/fs/ceph/file.c
+++ b/fs/ceph/file.c
@@ -632,7 +632,7 @@ ceph_sync_direct_write(struct kiocb *ioc
 					    false);
 		if (IS_ERR(req)) {
 			ret = PTR_ERR(req);
-			goto out;
+			break;
 		}
 
 		num_pages = calc_pages_for(page_align, len);
@@ -750,7 +750,7 @@ static ssize_t ceph_sync_write(struct ki
 					    false);
 		if (IS_ERR(req)) {
 			ret = PTR_ERR(req);
-			goto out;
+			break;
 		}
 
 		/*
